pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -137,6 +137,8 @@ type ShootSpec struct {
 	// This field is immutable.
 	// +optional
 	SchedulerName *string `json:"schedulerName,omitempty" protobuf:"bytes,21,opt,name=schedulerName"`
+	// CloudProfile is a reference to a CloudProfile or a NamespacedCloudProfile.
+	CloudProfile *CloudProfileReference `json:"cloudProfile,omitempty" protobuf:"bytes,22,req,name=cloudProfile"`
```

Comment:
Are there plans to deprecate the `CloudProfileName` field in favour of this one?
We can make the Gardener APIServer migrate the new field for existing clusters.

Reply:
Yes there are plans to perform a migration, but not as of this PR. Please see the other discussion about the `spec.CloudProfile` field for more information.